contrib/CMakeLists.txt

```diff
@@ -345,76 +350,3 @@ endif()
 if (USE_S2_GEOMETRY)
     add_subdirectory(s2geometry-cmake)
 endif()
-
-# Put all targets defined here and in subdirectories under "contrib/<immediate-subdir>" folders in GUI-based IDEs.
-# Some of third-party projects may override CMAKE_FOLDER or FOLDER property of their targets, so they would not appear
-# in "contrib/..." as originally planned, so we workaround this by fixing FOLDER properties of all targets manually,
-# instead of controlling it via CMAKE_FOLDER.
-
-function (ensure_target_rooted_in _target _folder)
-    # Skip INTERFACE library targets, since FOLDER property is not available for them.
-    get_target_property (_target_type "${_target}" TYPE)
-    if (_target_type STREQUAL "INTERFACE_LIBRARY")
-        return ()
-    endif ()
```

Comment:
This can be changed just to this to fix the issue:
```cmake
    # Skip aliases and INTERFACE library targets, since FOLDER property is not available for them.
    get_target_property (_target_aliased "${_target}" ALIASED_TARGET)
    get_target_property (_target_type "${_target}" TYPE)
    if (_target_aliased OR _target_type STREQUAL "INTERFACE_LIBRARY")
        return ()
    endif ()
```